### DISCUSSION TITLE
6.6.0 issue

### DISCUSSION THREAD

**Discussion by tanguy-e** [jgromes/RadioLib#1119] [Q&A] [answered]
Hello, it seems 6.6.0 version modifiy LORAWAN session activation but the radiolib-persistent examples are not updated to take into account this modification ? 
Thanks

**Comment by StevenCellist** [comment, marked as answer]
You are right - there are open PRs that I still have to review. I have some working adaptations locally as well, but am currently very busy finishing some assignments for uni. You may find that the PRs on the persistence library work well enough, but no guarantee that they cover some less frequent edge cases.
I hope to get to it around the weekend.

**Reply by StevenCellist** [reply]
@tanguy-e not exactly weekend anymore, but the example for ESP32 just got updated :)

**Comment by tanguy-e** [comment]
So i tried the new ESP32 radiolib-persistent example. i just changed the LoRaWAN parameters in the config.h. I use a HELTEC WiFi LoRa 32 V3 and it seems that the board is not recognised. The file says to see the notes for more info about this but in the notes "Pinmap How-To" is empty.

```
In file included from C:\Users\tanguy-e\Documents\Arduino\LoRaWAN_ESP32\LoRaWAN_ESP32.ino:34:
C:\Users\tanguy-e\Documents\Arduino\LoRaWAN_ESP32\config.h:89:67: note: '#pragma message: Unknown board - no automagic pinmap available'
   89 |   #pragma message ("Unknown board - no automagic pinmap available")
      |                                                                   ^
C:\Users\tanguy-e\Documents\Arduino\LoRaWAN_ESP32\config.h:107:19: error: 'radio' was not declared in this scope
  107 | LoRaWANNode node(&radio, &Region, subBand);
      |                   ^~~~~
C:\Users\tanguy-e\Documents\Arduino\LoRaWAN_ESP32\LoRaWAN_ESP32.ino: In function 'void setup()':
C:\Users\tanguy-e\Documents\Arduino\LoRaWAN_ESP32\LoRaWAN_ESP32.ino:173:11: error: 'radio' was not declared in this scope
  173 |   state = radio.begin();
      |           ^~~~~

exit status 1

Compilation error: 'radio' was not declared in this scope
```

**Comment by StevenCellist** [comment]
You are likely best off defining it yourself for now, there is a new system in the making. I am not going to troubleshoot the existing, so I recommend copying or defining the appropriate board/radio yourself.

**Reply by tanguy-e** [reply]
The card seemed to be supported so i asked the question ... Find a workaround

**Reply by StevenCellist** [reply]
Sorry, didn't mean to sound rude, the answer was just very very straight to the point :) let me know in case you have any problems!

**Comment by HeadBoffin** [comment]
As this board is the one we use the most for testing, this is a somewhat "interesting" situation - but we use PIO and it appears you are using Arduino, can't tell what version of IDE. Let me check the definition of `ARDUINO_heltec_wifi_lora_32_V3` in Arduino again.

**Reply by StevenCellist** [reply]
Had a look yesterday and there is a .variant name and a .board name, one of which is named wifi_lora_32_V3 and one of which wifi_32_lora_v3, where PIO uses lower case for both but Arduino uses CAPS for the latter but not the former... but I'm not setup for Arduino currently so didn't end up solving the problem. Proper mess...

**Comment by tanguy-e** [comment]
I'm using 2.3.2 Arduino IDE

**Comment by HeadBoffin** [comment]
Transpires that Arduino uses `ARDUINO_heltec_wifi_32_lora_V3` and PIO uses `ARDUINO_heltec_wifi_lora_32_V3` - go figure - and there are also references in the Arduino output thus: `-DARDUINO_BOARD="heltec_wifi_32_lora_V3" -DARDUINO_VARIANT="heltec_wifi_lora_32_V3"`

This looks like classic bit rot / poor attention to detail that inevitably turns up in big eco systems like Arduino & PlatformIO - rather like the rapid bit rot my original config.h scheme suffered when untested contributions were committed which seem to have leaked over to the persistence example. 

I'd not that the scheme was a quantum leap of automagic, previously you had to figure out the pin maps - either by search or using the new standard software development technique of asking someone to give the quick answer. The #pragma message alerted you what needed resolving. It is unfortunate that the volunteers haven't completed the instructions on how to read a schematic to get the pin map.

If you change the config.h thus it will pickup the pin map for either Arduino or PIO.

`#elif defined(ARDUINO_heltec_wifi_32_lora_V3) || defined(ARDUINO_heltec_wifi_lora_32_V3) `

I've done a smoke test, ie yes, it ~blends~ compiles, but haven't got to flash a board as they are in use right now. The whole setup is undergoing revision so I'll take a peek at progress & arrange for more comprehensive testing in the coming weeks.

**Comment by tanguy-e** [comment]
Changing 
```
#elif defined(ARDUINO_heltec_wifi_32_lora_V3) || defined(ARDUINO_heltec_wifi_lora_32_V3) 
 #pragma message ("Using Heltec WiFi LoRa32 v3 - Display + USB-C")
  SX1262 radio = new Module(8, 14, 12, 13);
```
leads to 
```
In file included from C:\Users\tanguy-e\Documents\Arduino\LoRaWAN_ESP32\LoRaWAN_ESP32.ino:34:
C:\Users\tanguy-e\Documents\Arduino\LoRaWAN_ESP32\config.h:89:67: note: '#pragma message: Unknown board - no automagic pinmap available'
   89 |   #pragma message ("Unknown board - no automagic pinmap available")
      |                                                                   ^
C:\Users\tanguy-e\Documents\Arduino\LoRaWAN_ESP32\config.h:110:19: error: 'radio' was not declared in this scope
  110 | LoRaWANNode node(&radio, &Region, subBand);
      |                   ^~~~~
C:\Users\tanguy-e\Documents\Arduino\LoRaWAN_ESP32\LoRaWAN_ESP32.ino: In function 'void setup()':
C:\Users\tanguy-e\Documents\Arduino\LoRaWAN_ESP32\LoRaWAN_ESP32.ino:193:11: error: 'radio' was not declared in this scope
  193 |   state = radio.begin();
      |           ^~~~~

exit status 1

Compilation error: 'radio' was not declared in this scope
```

**Reply by HeadBoffin** [reply]
Are you using the main ESP32 BSP or have you installed something else / extra?

Can you include the full compile output (set in preferences) as a text file attachment.

**Comment by tanguy-e** [comment]
I'm using ESP32 BSP from espressif in 3.0.1 version
Not sure how to find the full compile output ...

**Comment by tanguy-e** [comment]
See!
[compile.txt](https://github.com/user-attachments/files/15902548/compile.txt)

**Comment by HeadBoffin** [comment]
>Not sure how to find the full compile output ...

I said "set in preferences" - please read ALL of the answers given so we can help you faster.

I can't / won't test v3 as I've too much going on at present to destroy my business by trying out a whole new BSP that has shifted the goal post, so probably easiest to copy the relevant pin map in to the unknown section below it and crack on. 

Or you could take the totally massive hint given above with the change that I suggested above and put in the define that's now being used.

**Comment by tanguy-e** [comment]
I answered too quickly but i posted the file asked. I already have a workaround. It was just to help to debug this board problem ...

**Reply by HeadBoffin** [reply]
>  I already have a workaround. It was just to help to debug this board problem

Well then say so

**Comment by StevenCellist** [comment]
To solve the problem, it looks like the following will be your solution, judging by the compile log:
```
#elif defined(ARDUINO_heltec_wifi_32_lora_V3) || defined(ARDUINO_heltec_wifi_lora_32_V3) || defined(ARDUINO_HELTEC_WIFI_LORA_32_V3) 
```
The second one may be redundant, but at this point I'm not sure anymore whether the world is round or flat. Looks like Arduino and PIO are thinking it's a torus or something.

**Comment by tanguy-e** [comment]
Yes it works fine.
Thanks
